java/ql/test/utils/model-generator/typebasedflow/p/Stream.java

```diff
@@ -0,0 +1,249 @@
+    public <R> Stream<R> map(Function<? super T, ? extends R> mapper) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;mapToDouble;(ToDoubleFunction);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public DoubleStream mapToDouble(ToDoubleFunction<? super T> mapper) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;mapToInt;(ToIntFunction);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public IntStream mapToInt(ToIntFunction<? super T> mapper) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;mapToLong;(ToLongFunction);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public LongStream mapToLong(ToLongFunction<? super T> mapper) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;max;(Comparator);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    // MaD=p;Stream;true;max;(Comparator);;Argument[-1].Element;Argument[0].Parameter[1];value;generated
+    // MaD=p;Stream;true;max;(Comparator);;Argument[-1].Element;ReturnValue.SyntheticField[ArgType0];value;generated
+    public Optional<T> max(Comparator<? super T> comparator) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;min;(Comparator);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    // MaD=p;Stream;true;min;(Comparator);;Argument[-1].Element;Argument[0].Parameter[1];value;generated
+    // MaD=p;Stream;true;min;(Comparator);;Argument[-1].Element;ReturnValue.SyntheticField[ArgType0];value;generated
+    public Optional<T> min(Comparator<? super T> comparator) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;noneMatch;(Predicate);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public boolean noneMatch(Predicate<? super T> predicate) {
+        return false;
+    }
+
+    // Issue with model generator. ... is not supported.
+    // public static <T> Stream<T> of(T... t) {
+    // return null;
+    // }
+
+    // MaD=p;Stream;true;of;(Object);;Argument[0];ReturnValue.Element;value;generated
+    public static <T> Stream<T> of(T t) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;peek;(Consumer);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    // MaD=p;Stream;true;peek;(Consumer);;Argument[-1].Element;ReturnValue.Element;value;generated
+    public Stream<T> peek(Consumer<? super T> action) {
+        return null;
+    }
+
+    // Model generator yields a couple of extra results as models are generated for
+    // writing to the stream.
+    // MaD=p;Stream;true;reduce;(BinaryOperator);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    // MaD=p;Stream;true;reduce;(BinaryOperator);;Argument[-1].Element;Argument[0].Parameter[1];value;generated
+    // MaD=p;Stream;true;reduce;(BinaryOperator);;Argument[-1].Element;ReturnValue.SyntheticField[ArgType0];value;generated
+    // MaD=p;Stream;true;reduce;(BinaryOperator);;Argument[0].ReturnValue;Argument[-1].Element;value;generated
```

Comment:
Yeah, this one is spurious but mostly harmless. Maybe it would be nice if the regex for matching `MaD=` would allow a `SPURIOUS` prefix so we could write
```suggestion
    // SPURIOUS MaD=p;Stream;true;reduce;(BinaryOperator);;Argument[0].ReturnValue;Argument[-1].Element;value;generated
```

Reply:
Also: Argument[0].ReturnValue -> Argument[0].Parameter[1] is spurious. In the documentation it is stated that only required is that the operator is associative.

Reply:
Hmm - looks like that one is included in the manual models.